jdaviz/configs/mosviz/plugins/viewers.py

```diff
@@ -28,7 +28,11 @@ def set_plot_axes(self):
         data = self.data()[0]
         # Set axes labels for the spectrum viewer
 
-        spectral_axis_unit_type = data.spectral_axis.unit.physical_type.title()
+        try:
+            spectral_axis_unit_type = data.spectral_axis.unit.physical_type.title()
+        except AttributeError:
```

Comment:
`except` is expensive. Better to check for `astropy` version on import and use `if ... else ...` based on that check.

Reply:
I also wonder if `list(data.spectral_axis.unit.physical_type)[0].title()` would be better, in case you are accessing a unit with multiple entries, but if that is not a possibility, then you don't have to.